java/code/src/com/suse/manager/webui/services/SaltServerActionService.java

```diff
@@ -1481,6 +1481,27 @@ private Map<LocalCall<?>, List<MinionSummary>> imageInspectAction(
         }
     }
 
+    private String getChannelUrl(MinionServer minion, String channelLabel) {
+        DownloadTokenBuilder tokenBuilder = new DownloadTokenBuilder(minion.getOrg().getId());
+        tokenBuilder.useServerSecret();
+        tokenBuilder.setExpirationTimeMinutesInTheFuture(
+            Config.get().getInt(ConfigDefaults.TEMP_TOKEN_LIFETIME)
+        );
+        tokenBuilder.onlyChannels(Collections.singleton(channelLabel));
+        String t = "";
+        try {
+            t = tokenBuilder.getToken();
+        }
+        catch (JoseException e) {
+            e.printStackTrace();
+        }
+        final String token = t;
+
+        String host = minion.getChannelHost();
+
+        return "https://" + host + "/rhn/manager/download/" + channelLabel + "?" + token;
```

Comment:
I know this is just copy/refactor, but I don't get it.
We have a t and then pass it to token. Why not just use one variable?
Also with this exception handling we just pollute logs with out of nowhere stack traces. Shouldn't we just rethrow?

Reply:
I have fixed the variable.
For the exception I am not sure, when it appears and how to handle it better.